WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -399,6 +399,11 @@ public async Task ReadyToSignAsync(ReadyToSignRequestRequest request, Cancellati
 				}
 			}
 
+			if (alice.ReadyToSign)
+			{
+				throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.AlreadySigned);
+			}
```

Comment:
I think it makes sense to start with noting the input, since we no longer have a `Ban` method and the interaction with blame round creation needs to be handled anyway.

```suggestion
			if (alice.ReadyToSign)
			{
				Prison.Note(alice, round.Id);
				throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.AlreadySigned);
			}
```

Reply:
The `Prision::Ban` method: https://github.com/molnard/WalletWasabi/blob/50d6fc602d3c75e6093e38a1994856c3ff26a9a9/WalletWasabi/WabiSabi/Backend/Banning/Prison.cs#L45-L48

Reply:
If we go straight to banning I think that will do the right thing with the current blame round handling. Although the input will be allowed, input registration should fail, and it will be as if the input wasn't registered. There's only an edge case where the banned input's removal takes the allowed prevout list below the minimum threshold, which would mean the blame round can only time out, we can cover that in a followup PR with a test.

Reply:
I added it here.